### PR TITLE
removing old deprecations and Optimise module

### PR DESCRIPTION
**Pull request by CarloLucibello**
Last part of #1986

### CODE REVIEW

File: docs/src/training/optimisers.md [hunk 1]
```diff
@@ -4,76 +4,63 @@ CurrentModule = Flux
...
 
 ```julia
-opt = Optimiser(ExpDecay(1, 0.1, 1000, 1e-4), Descent())
+opt = OptimiserChain(WeightDecay(1e-4), Descent())
 ```
 
-Here we apply exponential decay to the `Descent` optimiser. The defaults of `ExpDecay` say that its learning rate will be decayed every 1000 steps.
-It is then applied like any optimiser.
+Here we apply the weight decay to the `Descent` optimiser. 
+The resultin optimser `opt` can be used as any optimiser.
```
Comment (christiangnrd) [inline docs/src/training/optimisers.md:44]:
```suggestion
The resulting optimiser `opt` can be used as any optimiser.
```
Typos

File: docs/src/training/reference.md [hunk 2]
```diff
@@ -60,8 +60,8 @@ See the [Optimisers documentation](https://fluxml.ai/Optimisers.jl/dev/) for det
 
 ```@docs
 Flux.params
-Flux.Optimise.update!(opt::Flux.Optimise.AbstractOptimiser, xs::AbstractArray, gs)
-Flux.Optimise.train!(loss, ps::Flux.Params, data, opt::Flux.Optimise.AbstractOptimiser; cb)
+Flux.update!(opt::Flux.Optimise.AbstractOptimiser, xs::AbstractArray, gs)
+Flux.train!(loss, ps::Flux.Params, data, opt::Flux.Optimise.AbstractOptimiser; cb)
```
Comment (christiangnrd) [inline docs/src/training/reference.md:64]: References to `Flux.Optimise.AbstractOptimiser` remaining.

File: src/deprecations.jl [hunk 3]
```diff
@@ -1,225 +1,13 @@
...
-  Dense(in => out, σ; kw...)
-Bilinear(in1::Integer, in2::Integer, out::Integer, σ = identity; kw...) =
-  Bilinear((in1, in2) => out, σ; kw...)
-Embedding(in::Integer, out::Integer; kw...) = Embedding(in => out; kw...)
-
-RNNCell(in::Integer, out::Integer, σ = tanh; kw...) = RNNCell(in => out, σ; kw...)
-LSTMCell(in::Integer, out::Integer; kw...) = LSTMCell(in => out; kw...)
-
-GRUCell(in::Integer, out::Integer; kw...) = GRUCell(in => out; kw...)
-GRUv3Cell(in::Integer, out::Integer; kw...) = GRUv3Cell(in => out; kw...)
```
Comment (mcabbott) [inline src/deprecations.jl:57]: Most of this would be great to delete!

These lines though, I wonder if we should keep. Or change them to print a depwarn.

**Comment by CarloLucibello** [comment]
closing in favor of #2495

**Pull request closed without merging**